### PR TITLE
Revert #3291

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary
- Ticket no. 141406
- Revert https://github.com/openvinotoolkit/training_extensions/pull/3291 and upgrade `jsonargparse` to `4.28.0`.
- Remove `loss_callable` from cls models due to errors on pickling during hpo https://github.com/openvinotoolkit/training_extensions/actions/runs/9110533591/job/25047364781

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sungchul1 — APPROVED** [review]

**Review by eunwoosh — APPROVED** [review]

File: src/otx/recipe/classification/h_label_cls/deit_tiny.yaml [hunk 1]
```diff
@@ -3,16 +3,6 @@ model:
   init_args:
     arch: "deit-tiny"
 
-    multiclass_loss_callable:
-      class_path: torch.nn.CrossEntropyLoss
```
Comment (vinnamkim) [inline src/otx/recipe/classification/h_label_cls/deit_tiny.yaml:7]: @harimkang
Why is it `reduction="mean"` (default of `torch.nn.CrossEntropyLoss`) although multiclass cls model has `reduction="none"`?

File: src/otx/recipe/classification/multi_class_cls/deit_tiny.yaml [hunk 2]
```diff
@@ -4,11 +4,6 @@ model:
     label_info: 1000
     arch: "deit-tiny"
 
-    loss_callable:
-      class_path: torch.nn.CrossEntropyLoss
-      init_args:
-        reduction: none
```
Comment (vinnamkim) [inline src/otx/recipe/classification/multi_class_cls/deit_tiny.yaml:10]: @harimkang

File: src/otx/recipe/classification/h_label_cls/deit_tiny.yaml (hunk 1 above)
Comment (harimkang) [reply src/otx/recipe/classification/h_label_cls/deit_tiny.yaml:7]: Um I remember it being imported as it was used in MM, some calculations were used differently in head.

File: src/otx/recipe/classification/multi_class_cls/deit_tiny.yaml (hunk 2 above)
Comment (harimkang) [reply src/otx/recipe/classification/multi_class_cls/deit_tiny.yaml:10]: Thanks to share. I'll take a look at it later to see if there are any potential issues.

**Review by harimkang — APPROVED** [review]

File: src/otx/recipe/classification/multi_class_cls/deit_tiny.yaml (hunk 2 above)
Comment (vinnamkim) [reply src/otx/recipe/classification/multi_class_cls/deit_tiny.yaml:10]: Please create a JIRA ticket to clean up those things.

**Pull request merged**